app/models/secret_storage.rb

```diff
@@ -80,7 +80,7 @@ def delete(key)
 
       def keys
         keys = vault_client.list(VAULT_SECRET_BACKEND + SAMSON_SECRET_NAMESPACE)
-        keys = keys_recursive(keys)
+        keys = keys_recursive(keys).uniq # flatten all the keys since the pod/vault association is included now
```

Comment:
`# we read from multiple backends that might have the same keys` ... can also call `keys.uniq!` in a separate line to avoid a bit of overhead ...